libcaf_test/caf/test/runnable.hpp

```diff
@@ -65,7 +65,13 @@ class CAF_TEST_EXPORT runnable {
                 const detail::source_location& location
                 = detail::source_location::current()) {
     assert_save_comparison<T0, T1>();
-    if (lhs == rhs) {
+    if constexpr (std::is_floating_point_v<T0>
+                  && std::is_floating_point_v<T1>) {
+      if (lhs - rhs < std::numeric_limits<T0>::epsilon()) {
+        reporter::instance().pass(location);
+        return true;
+      }
+    } else if (lhs == rhs) {
```

Comment:
I think the precision should not be hidden away in a check function. Instead, let's extend `assert_save_comparison` to prohibit floating points comparison using `operator==`.

I like the approach of doctest, so let's do something similar with an explicit `approx` class like this:

```
/// Allows comparing a value of type `T` with a configurable precision.
template <class T>
class approx {
public:
  explicit approx(T value) : value_(value) { }

  approx epsilon(T eps) const {
    auto copy = *this;
    copy.epsilon_ = eps;
    return copy;
  }

  friend class operator==(T lhs, approx<T> rhs) {
    return lhs >= min_value() && lhs <= max_value();
  }

  // ... operator==(approx<T>, T) ...
private:
  T min_value() { return value_ - epsilon_; }
  T max_value() { return value_ + epsilon_; }
  T value_;
  T epsilon_ = std::numeric_limit<T>::epsilon();
};
```

This will allow users to write `check_eq(my_value, approx{3.127})` to get the default precision or `check_eq(my_value, approx{3.127}.epsilon(0.001))` to do the comparison with a custom precision.